respect-app-compose/src/commonMain/kotlin/world/respect/app/view/clazz/list/ClazzListScreen.kt

```diff
@@ -0,0 +1,116 @@
+package world.respect.app.view.clazz.list
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.unit.dp
+import world.respect.app.app.RespectAsyncImage
+import world.respect.app.components.RespectSortOption
+import world.respect.shared.viewmodel.clazz.list.ClazzListUiState
+import world.respect.shared.viewmodel.clazz.list.ClazzListViewModel
+import world.respect.shared.viewmodel.clazz.list.ClazzListViewModel.Companion.NAME
+
+@Composable
+fun ClazzListScreen(
+    viewModel: ClazzListViewModel
+) {
+
+    val uiState by viewModel.uiState.collectAsState()
+
+    ClazzListScreen(
+        uiState = uiState,
+        onClickClazz = { viewModel.onClickClazz() },
+        onClickFilter = { viewModel.onClickFilter(it) },
+    )
+}
+
+@Composable
+fun ClazzListScreen(
+    uiState: ClazzListUiState,
+    onClickClazz: () -> Unit,
+    onClickFilter: (String) -> Unit,
+) {
+
+    Column(
+        modifier = Modifier.fillMaxSize().padding(16.dp)
+    ) {
+
+        RespectSortOption(
+            options = uiState.sortOptions,
+            selectedOption = uiState.selectedSortOption ?: NAME,
+            onOptionSelected = onClickFilter,
+            optionLabel = {it},
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp)
+        )
+
+        LazyColumn(
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+        ) {
+
+            itemsIndexed(
+                //dummy list
```

Comment:
Whether or dummy data or real data **should have no effect on the UI or ViewModel**